[CI] ForecastIT testOverflowToDisk failing

**Build Scans:**
- [elasticsearch-periodic #5111 / graalvm-ce17_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/4xma5jnp65wnq)
- [elasticsearch-periodic #5087 / graalvm-ce17_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/6vpiwmhfkqlvs)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.ml.integration.ForecastIT.testOverflowToDisk" -Dtests.seed=3BB615DC337D87CB -Dtests.locale=bg-BG -Dtests.timezone=America/Cambridge_Bay -Druntime.java=17
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.ForecastIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testOverflowToDisk),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: one of {<finished>}
     but: was <scheduled>
```

**Issue Reasons:**
- [8.x] 2 failures in test testOverflowToDisk (1.3% fail rate in 150 executions)
- [8.x] 2 failures in step graalvm-ce17_checkpart4_java-matrix (40.0% fail rate in 5 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (40.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -94,9 +94,6 @@ tests:
 - class: org.elasticsearch.smoketest.MlWithSecurityIT
   method: test {yaml=ml/sparse_vector_search/Test sparse_vector search with query vector and pruning config}
   issue: https://github.com/elastic/elasticsearch/issues/119548
-- class: org.elasticsearch.xpack.ml.integration.ForecastIT
-  method: testOverflowToDisk
-  issue: https://github.com/elastic/elasticsearch/issues/117740
 - class: org.elasticsearch.multi_cluster.MultiClusterYamlTestSuiteIT
   issue: https://github.com/elastic/elasticsearch/issues/119983
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeAutodetectIntegTestCase.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeAutodetectIntegTestCase.java
@@ -246,7 +246,19 @@ abstract class MlNativeAutodetectIntegTestCase extends MlNativeIntegTestCase {
 
     protected void waitForecastToFinish(String jobId, String forecastId) throws Exception {
         // Forecasts can take an eternity to complete in the FIPS JVM
-        waitForecastStatus(inFipsJvm() ? 300 : 90, jobId, forecastId, ForecastRequestStats.ForecastRequestStatus.FINISHED);
+        int timeoutSeconds = inFipsJvm() ? 300 : 90;
+        // First wait for the forecast document to exist and be in a non-terminal state
+        // This handles the race condition where the document may be SCHEDULED or STARTED initially
+        waitForecastStatus(
+            timeoutSeconds,
+            jobId,
+            forecastId,
+            ForecastRequestStats.ForecastRequestStatus.SCHEDULED,
+            ForecastRequestStats.ForecastRequestStatus.STARTED,
+            ForecastRequestStats.ForecastRequestStatus.FINISHED
+        );
+        // Then wait specifically for FINISHED status
+        waitForecastStatus(timeoutSeconds, jobId, forecastId, ForecastRequestStats.ForecastRequestStatus.FINISHED);
     }
 
     protected void waitForecastStatus(String jobId, String forecastId, ForecastRequestStats.ForecastRequestStatus... status)
@@ -261,6 +273,10 @@ abstract class MlNativeAutodetectIntegTestCase extends MlNativeIntegTestCase {
         ForecastRequestStats.ForecastRequestStatus... status
     ) throws Exception {
         assertBusy(() -> {
+            // Refresh the index to ensure recently indexed forecast stats documents are visible
+            indicesAdmin().prepareRefresh(AnomalyDetectorsIndex.jobResultsAliasedName(jobId))
+                .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_HIDDEN)
+                .get();
             ForecastRequestStats forecastRequestStats = getForecastStats(jobId, forecastId);
             assertThat(forecastRequestStats, is(notNullValue()));
             assertThat(forecastRequestStats.getStatus(), in(status));

